DoSomethingArchive/legacy-website
# Campaign closed template setup

## Closed template / Campaign Run content
- Adds a menu callback for Campaign closed state: `node/%node/closed`
- Adds a function `dosomething_campaign_is_closed($node)` to determine if a campaign should be rendered in the closed state.   Fixes #2351 .
  - Refs #2377 - For now, `dosomething_campaign_is_closed` simply returns TRUE when a staff member is viewing the node at the Closed page: `node/%node/closed` as we build out the content type and template.
- Adds new tpl `node--campaign--closed.tpl.php` for a Campaign's closed state.  Adds condition into `dosomething_campaign_preprocess_node` to use the template when `dosomething_campaign_is_closed($node)`. 
- Adds a function `dosomething_campaign_run_preprocess_closed`, which will lookup the Campaign Run node and preprocess the relevant varables to display in Campaign's closed template. Fixes #2352 
### Test:
- `ds pull stage`
- Pull this branch down
- `drush -y fra`
- Create a new Campaign Run node, and select Comeback Clothes. Fill out content and save.
- Visit `node/362/closed`, confirm the closed template is displayed.
- Confirm STAFF only can access a node's closed page at `node/362/closed`
  - Access denied to anonymous and authenticated non-staff users.  Actual logic for when to display as default for the node's content TBD in #2377
## Pitch Page fix
- Fixes the staff-only pitch page callback per changes made in #2383. When the `pitch` view mode was removed, this page callback needed to be updated as well.
### Test:
- Confirm STAFF only can access a node's pitch page at `node/*/pitch`
  - Access denied to anonymous and authenticated non-staff users.
- Confirm it's the Pitch Page content
